lib/fluent/plugin/in_tail.rb

```diff
@@ -537,86 +535,144 @@ def swap_state(pe)
         mpe = MemoryPositionEntry.new
         mpe.update(pe.read_inode, pe.read_pos)
         @pe = mpe
-        @io_handler.pe = mpe # Don't re-create IOHandler because IOHandler has an internal buffer.
-
         pe # This pe will be updated in on_rotate after TailWatcher is initialized
       end
 
       class StatWatcher < Coolio::StatWatcher
-        def initialize(path, log, &callback)
+        def initialize(watcher, &callback)
```

Comment:
Other helper classes needed a similar change for constructor parameters, and it made sense for me to apply the same policy to `StatWatcher` as well.